src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md

```diff
@@ -1,4 +1,400 @@
 ---
 title: 'k6 Extensions'
-redirect: 'https://k6.io/blog/extending-k6-with-xk6'
----
\ No newline at end of file
+---
+
+> *Since v0.29.0 (JavaScript extensions) and v0.31.0 (Output extensions)*
+
+Traditionally, extending k6 with custom functionality that isn't available in the
+open source tool has been possible in one of two ways:
+
+- By importing a JavaScript library. This is a simple way to extend the functionality
+  of a test script, though it has some drawbacks. Since JavaScript in k6 runs in a
+  virtual machine [that is unlike the one used in web browsers and
+  NodeJS](/#what-k6-does-not), it's not possible to use libraries that write to
+  disk, use JS APIs other than the ones built into k6, or need some new binary
+  protocol support. Furthermore, while k6 performs well in most use cases, all
+  execution is interpreted by the [Go JavaScript runtime](https://github.com/dop251/goja),
+  which can impact the performance of certain resource intensive operations
+  (cryptography, compression, etc.).
+
+- By forking the [k6 repository](https://github.com/k6io/k6), doing the changes in Go
+  and submitting a pull request for review by the core team (preferably following
+  the [contribution guidelines](https://github.com/k6io/k6/blob/master/CONTRIBUTING.md)).
+  This is a great way to contribute new core features, but it can be a lengthy
+  process, and submissions might be rejected if they don't align with the long-term
+  vision of the project.
+
+To address these issues and allow the community to more easily adapt k6 to fit their
+needs, [k6 v0.29.0](https://github.com/k6io/k6/releases/tag/v0.29.0) introduced
```

Comment:
imo, it is unnecessary to mention again the 0.29 / 0.31 versions

Reply:
OK, done in 0ec0f65.